src/adapters/mc/Baseline.h

```diff
@@ -9,14 +9,14 @@ extern "C"
 {
 #endif
 
-int BaselineIsValidResourceIdRuleId(const char* resourceId, const char* ruleId, const char* payloadKey, OsConfigLogHandle log);
-int BaselineIsCorrectDistribution(const char* payloadKey, OsConfigLogHandle log);
+int BaselineIsValidResourceIdRuleId(const char* resourceId, const char* ruleId, const char* payloadKey, OsConfigLogHandle log, OSConfigTelemetryHandle telemetry);
+int BaselineIsCorrectDistribution(const char* payloadKey, OsConfigLogHandle log, OSConfigTelemetryHandle telemetry);
```

Comment:
> OSConfigTelemetryHandle telemetry) [](http://example.com/codeflow?start=80&length=35)

Traditionally, Telemetry is not like Logging. When adding Telemetry to a project, you do that with the least amount of churn. 

Furthermore, we only have 1 telemetry client for entire OSConfig instance. Logs are per each OSConfig module. 

Instead to go the logging handle way and modify all major functions in project and code everywhere, try this:

Declare a global telemetry handle inside of the client library and then internally reference that from all telemetry calls.

*You can do this and do not modify code beyond adding the telemetry event writes*